File: modules/admin-ui/src/main/resources/web/admin/common/js/module/ModuleBasedName.ts
module api.module {


    export class ModuleBasedName implements api.Equitable {
        static SEPARATOR = ":";

        private refString: string;

        private applicationKey: ApplicationKey;

        private localName: string;

        constructor(applicationKey: ApplicationKey, localName: string) {
            this.applicationKey = applicationKey;
            this.localName = localName;
            this.refString = applicationKey.toString() ? applicationKey.toString() + ModuleBasedName.SEPARATOR + localName : localName;
        }

        getLocalName(): string {
            return this.localName;
        }

        getApplicationKey(): ApplicationKey {
            return this.applicationKey;
        }

        toString(): string {
            return this.refString;
        }

        equals(o: api.Equitable): boolean {

            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, api.ClassHelper.getClass(this))) {
                return false;
            }

            var other = <ModuleBasedName>o;

            if (!api.ObjectHelper.stringEquals(this.refString, other.refString)) {
                return false;
            }

            return true;
        }

    }

}

File: modules/core-api/src/main/java/com/enonic/xp/module/ModuleBasedName.java
package com.enonic.xp.module;


import com.google.common.base.Joiner;

public abstract class ModuleBasedName
{
    protected final static String SEPARATOR = ":";

    private final String refString;

    private final ModuleKey moduleKey;

    private final String localName;

    protected ModuleBasedName( final ModuleKey moduleKey, final String localName )
    {
        this.moduleKey = moduleKey;
        this.localName = localName;
        this.refString = Joiner.on( SEPARATOR ).join( this.moduleKey.toString(), this.localName );
    }

    public String getLocalName()
    {
        return localName;
    }

    public ModuleKey getModuleKey()
    {
        return moduleKey;
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( o == null || getClass() != o.getClass() )
        {
            return false;
        }

        final ModuleBasedName moduleBasedName = (ModuleBasedName) o;

        return refString.equals( moduleBasedName.refString );
    }

    @Override
    public int hashCode()
    {
        return refString.hashCode();
    }

    @Override
    public String toString()
    {
        return refString;
    }

}
